src/utils.js

```diff
@@ -351,7 +351,37 @@ export function computeLinesDefault({ctx, text, max, wordSpacing, letterSpacing}
     }
 
     if (part) {
-        parts.push(part);
+        const width = parseInt(ctx.measureText(part).width, 10);
+        let breakPointType = null;
+
+        if (options.wordBreak === 'break-all') {
+            breakPointType = 'BAI';
```

Comment:
As i said, the break-all case is already handled here: 
https://github.com/bezoerb/text-metrics/blob/master/src/index.js#L143
https://github.com/bezoerb/text-metrics/blob/master/src/utils.js#L426